### DISCUSSION TITLE
[Feature Request] multiple marker support in slider

### DISCUSSION THREAD

**Discussion by zerosoul** [heroui-inc/heroui#4685] [Feature requests]
### Is your feature request related to a problem? Please describe.

I am making a gradient online tool, and need a supporting multiple markers slider: https://sinqi.tools/pattern/gradient

![image](https://github.com/nextui-org/nextui/assets/2985895/746ca887-cb42-4181-a5b2-34b470208beb)

### Describe the solution you'd like

support multiple markers in slider

### Describe alternatives you've considered

demo here: https://cssgradient.io/

### Screenshots or Videos

![image](https://github.com/nextui-org/nextui/assets/2985895/87e54d7c-12d2-4189-98dc-5ce43e5df19d)

**Comment by FacundoTenuta** [comment]
You can do it. In the 'value' attribute, you can pass an array of values. Each marker is an element of the array. For example:

```
import { useState } from 'react';
import { Slider } from '@nextui-org/react';

export default function Home() {
	const [sliderValues, setSliderValues] = useState<number | number[]>([100, 500, 800]);

	return (
		<Slider
			label="Price Range"
			step={50}
			minValue={0}
			maxValue={1000}
			defaultValue={[100, 500, 900]}
			onChange={(value) => {
				setSliderValues(value);
			}}
			value={sliderValues}
			formatOptions={{ style: 'currency', currency: 'USD' }}
			className="max-w-md"
		/>
	);
}
```

Maybe this can help you. :D
<img width="556" alt="Captura de pantalla 2023-11-28 a la(s) 00 42 07" src="https://github.com/nextui-org/nextui/assets/18177807/29777516-27d8-41ce-90a6-846a85f745c9">

**Comment by zerosoul** [comment]
> You can do it. In the 'value' attribute, you can pass an array of values. Each marker is an element of the array. For example:
> 
> ```
> import { useState } from 'react';
> import { Slider } from '@nextui-org/react';
> 
> export default function Home() {
> 	const [sliderValues, setSliderValues] = useState<number | number[]>([100, 500, 800]);
> 
> 	return (
> 		<Slider
> 			label="Price Range"
> 			step={50}
> 			minValue={0}
> 			maxValue={1000}
> 			defaultValue={[100, 500, 900]}
> 			onChange={(value) => {
> 				setSliderValues(value);
> 			}}
> 			value={sliderValues}
> 			formatOptions={{ style: 'currency', currency: 'USD' }}
> 			className="max-w-md"
> 		/>
> 	);
> }
> ```
> 
> Maybe this can help you. :D <img alt="Captura de pantalla 2023-11-28 a la(s) 00 42 07" width="556" src="https://private-user-images.githubusercontent.com/18177807/286112848-29777516-27d8-41ce-90a6-846a85f745c9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNDU2ODgsIm5iZiI6MTcwMTI0NTM4OCwicGF0aCI6Ii8xODE3NzgwNy8yODYxMTI4NDgtMjk3Nzc1MTYtMjdkOC00MWNlLTkwYTYtODQ2YTg1Zjc0NWM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI5VDA4MDk0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0MjFhNzFkMzQ2ZTEzNmU0MDk4MTYxNTc1ZWNkY2ZhNzA3OTBlNmJkMmJkMjA5ZDRkZTNhNTgyMDAyYTljMGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MpLmyaS7Z24fE3tG82ji6TJWDHYGNBfJXUVsz9grJCI">

Thank you for the response. What I have in mind is the ability to dynamically add markers. Wherever the user clicks at slider, a new marker would be added at that location. Currently, I haven't figured out how to implement it yet.
